spec/models/article_spec.rb

```diff
@@ -708,6 +708,41 @@ def build_and_validate_article(*args)
     end
   end
 
+  describe "#main_image_from_frontmatter" do
+    it "set to true if markdown has cover_image" do
+      article = create(
+        :article,
+        user: user,
+        body_markdown: "---\ntitle: hey\npublished: false\ncover_image: #{Faker::Avatar.image}\n---\nYo",
+      )
+      expect(article.main_image_from_frontmatter).to eq(true)
+    end
+
+    context "when false" do
+      it "does not remove main image if cover image not passed in markdown" do
+        expect(article.main_image).not_to be_nil
+        article.update! body_markdown: "---\ntitle: hey\npublished: false\n---\nYo ho ho#{rand(100)}"
+        expect(article.reload.main_image).not_to be_nil
+      end
+
+      it "does remove main image if cover image is passed empty in markdown" do
+        expect(article.main_image).not_to be_nil
+        article.update! body_markdown: "---\ntitle: hey\npublished: false\ncover_image: \n---\nYo ho ho#{rand(100)}"
+        expect(article.reload.main_image).to be_nil
+      end
+    end
```

Comment:
These specs don't reference `main_image_from_frontmatter` at all, so they're relying on containing RSpec contexts. It seems we're assuming them to be `false` here because that's the default value if not provided and the containing contexts currently don't set them, but it's not guaranteed inside the specs in this context. If `article` changes elsewhere in this file to set `main_image_from_frontmatter` (either explicitly or implicitly), these specs could fail for unrelated reasons.

Could you explicitly set this value to `false` inside this context?